sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -223,8 +223,8 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks with PlanTestBa
           }
         } else {
           val lit = InternalRow(expected, expected)
-          val expectedRow =
-            UnsafeProjection.create(Array(expression.dataType, expression.dataType)).apply(lit)
+          val dtAsNullable = expression.dataType.asNullable
+          val expectedRow = UnsafeProjection.create(Array(dtAsNullable, dtAsNullable)).apply(lit)
```

Comment:
Can we add a test for this change?

Reply:
Thank you for your comment. Do you think what test is preferable?
* Successfully pass (we already have these UTs since all UTs have been passed)
* Expectedly failure (In other words, add a function that generated incorrect result)

Reply:
I'd like to add a test that will be failed without this patch.

Reply:
Since this patch allows us to correctly detect incorrect results, I added a test that was passed w/o this patch and that is failed with this patch.

Reply:
oh, yes. this is going to detect such failure.